Maui.RevenueCat.InAppBilling/Platforms/Android/Extensions/OfferingListExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿using Com.Revenuecat.Purchases;
+using Maui.RevenueCat.InAppBilling.Models;
+
+namespace Maui.RevenueCat.InAppBilling.Platforms.Android.Extensions;
+
+internal static class OfferingListExtensions
+{
+    public static IList<OfferingDto> ToOfferingDtoList(this Offerings offerings)
+    {
```

Comment:
I disagree on the ToList, if it was called ToList I would expect it to return List<Offerings> and not List<OfferingDto>. It doesn't seem consistent with the other extensions either. Are you sure about that?

Reply:
Can you explain why you avoid IList?  It makes these extensions more generic and also more testable in case we ever end up adding tests.  This isn't the only place I've added this in the project either, as its a typically considered a best practice...

Reply:
> 
> > I disagree on the ToList, if it was called ToList I would expect it to return List and not List. It doesn't seem consistent with the other extensions either. Are you sure about that?
> 
> You are right here, in case we keep it, it should probably be called `ToIList()` ?

I completely misstated my point on this one.  What I was trying to say is that if it was ToList(), I would expect it to return List<Offerings> but this actually returns List<OfferingDto>.  The default behavior of ToList() throughout. net is that it returns a List of the same object type and never does a conversion from one type to another.  This method does both a mapping from  OfferingDto to Offerings and converts it into a List.  This is why I chose the naming I did. 

I'm not sure attached though if you feel strongly let me know, just trying to be consistent with established .net norms.

Reply:
> * I just realized it shouldnt be public but internal and thus when we know final type the IList shouldnt be necessary but even if so, please try to answer questions above maybe after I comprehend some reasoning I agree with we can change it across the whole package.

I agree this should be internal, along with PackageListExtension, I made this change

Reply:
> I am probably not used to use it. But I have several question/concerns regarding this:
> 
> * How it will improve testability ? I personally dont see the benefit there

> * What benefits from consumer's POV does have IList ? I read some forum posts but opinions on this topic vary

Using interfaces encourages testability by creating architectural seams between different software components.  These seams can be used as a way to easily add unit tests to an application.  They also allow for dependency injection which has become the fundamental design philosophy of .net core to the point that it is one of the few things that is in the core of asp.net .  

There is a ton of info on this out there, when I google on the subject a bunch of it is from 2010 when interfaces were still a new concept in the Microsoft programming world.  I was going to paste something here but there is just so much I wouldn't know what to choose.

Ultimately it is an opinionated design pattern but seems to be the way that most .net is being written, especially internally at Microsoft.  I typically try to go with the flow of the community and use Microsoft's internal style as my guide.

In this case, I don't think from a consumers perspective it makes a lot of difference.  Now that it is marked internally the only consumer is this library itself.  If it was still public it would allow the consumer to implement their own IList to receive the contents which could be useful.  If we went and fixed the public interface to be an IList (which eliminates the need for the strange ToList()) then it would also be useful to the consumer because then they aren't forced to consume the results as a List but anything that implements the IList interface.

> * We are doing double allocation of the list(one internal and then external `ToList`), if we really decide to go this way, then probably using `yield return` would make sense instead of initializing `new List<OfferingDto>();` internally ?

I think if we go this way we change the public interface and make this a breaking change so that LoadOfferings returns IList instead of List.  Otherwise we should pass List around internally and eliminate the ToList.  The halfway point it is in now seems like a bad idea now that we are discussing it.

Reply:
Thank you for the explanation of `ToOfferingDtoList` and I agree, that if it is a mapping to different type then the name makes sense.

I agree with dependency injection and making input parameters more abstract (with use of interfaces) but only when it makes sense. But overall I am quite defensive against making things abstract without obvious benefit.
So after we have spotted that this thing is internal-only and we use lists across the library I would stick to simple `List`.
If in the future someone comes with a good usecase why he needs return collection more abstract and not just hardcoded `List` I would reconsider it across the whole library but until than we will keep it this way.

Reply:
Alright, good discussion.  I just made the adjustment.